tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/XUnit/XUnitTestOutputHelperQueueTestOutputIntegration.cs

```diff
@@ -58,52 +60,78 @@ internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, string
         private class XUnitLogEvent : DatadogLogEvent
         {
             private readonly string _message;
-            private readonly Context? _context;
+            private readonly Span _span;
 
             public XUnitLogEvent(string message, Span span)
             {
                 _message = message;
-                _context = span is null ? null : new Context(span.TraceId, span.SpanId, span.Context.Origin);
+                _span = span;
             }
 
             public override void Format(StringBuilder sb, LogFormatter formatter)
             {
-                formatter.FormatLog<Context?>(
-                    sb,
-                    _context,
-                    DateTime.UtcNow,
-                    _message,
-                    eventId: null,
-                    logLevel: DirectSubmissionLogLevelExtensions.Information,
-                    exception: null,
-                    (JsonTextWriter writer, in Context? state) =>
-                    {
-                        if (state.HasValue)
-                        {
-                            writer.WritePropertyName("dd.trace_id");
-                            writer.WriteValue($"{state.Value.TraceId}");
-                            writer.WritePropertyName("dd.span_id");
-                            writer.WriteValue($"{state.Value.SpanId}");
-                            writer.WritePropertyName("_dd.origin");
-                            writer.WriteValue($"{state.Value.Origin}");
-                        }
-
-                        return default;
-                    });
-            }
+                using var writer = LogFormatter.GetJsonWriter(sb);
 
-            private readonly struct Context
-            {
-                public readonly ulong TraceId;
-                public readonly ulong SpanId;
-                public readonly string Origin;
+                // Based on JsonFormatter
+                writer.WriteStartObject();
 
-                public Context(ulong traceId, ulong spanId, string origin)
+                writer.WritePropertyName("ddsource", escape: false);
+                writer.WriteValue("xunit");
+
+                if (HostMetadata.Instance.Hostname is { } hostname)
                 {
-                    TraceId = traceId;
-                    SpanId = spanId;
-                    Origin = origin;
+                    writer.WritePropertyName("hostname", escape: false);
+                    writer.WriteValue(hostname);
                 }
+
+                writer.WritePropertyName("timestamp", escape: false);
+                writer.WriteValue(DateTimeOffset.UtcNow.ToUnixTimeNanoseconds() / 1_000_000);
+
+                writer.WritePropertyName("message", escape: false);
+                writer.WriteValue(_message);
+
+                writer.WritePropertyName("status", escape: false);
+                writer.WriteValue("info");
+
+                string env = string.Empty;
+                if (_span is { } span)
```

Comment:
Are these required? e.g. `service` presumably is required even if there's no span?

Reply:
I did a workaround for `service` and `env` in https://github.com/DataDog/dd-trace-dotnet/pull/3037/commits/ecf89fa2a08dec1f5a2215b0aadff5a8fef81f2c, others are not required.